src/composable/composable.ts

```diff
@@ -67,15 +72,16 @@ function composable<T extends Fn>(fn: T): Composable<T> {
  * const d = C.pipe(a, b)
  * //    ^? Composable<({ aNumber }: { aNumber: number }) => { aBoolean: boolean }>
  */
-function pipe<T extends [Composable, ...Composable[]]>(...fns: T) {
+function pipe<T extends [Composable, ...Composable[]]>(
+  ...fns: T & PipeArguments<T>
```

Comment:
this trickery is what allows recursive checking of arguments type. `T` is evaluated first and it assumes the inferred type of given arguments, then TS will try to intersect with the result of `PipeArguments`.

Reply:
Ok, this explanation is what clicked for me! TY